site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -517,34 +570,48 @@ def _transform_additional_descriptions(self):
 
     def _search_vocabulary(self, term, vocab_type):
         """Search vocabulary utility function."""
+        inspire_id = self.inspire_metadata.get("control_number")
+        current_app.logger.debug(
+            f"[inspire_id={inspire_id}] Searching vocabulary '{vocab_type}' for term: '{term}'"
+        )
+
         service = current_service_registry.get("vocabularies")
         if "/" in term:
             # escape the slashes
+            current_app.logger.debug(
+                f"[inspire_id={inspire_id}] Escaping slashes in term: '{term}'"
+            )
             term = f'"{term}"'
         try:
             vocabulary_result = service.search(
                 system_identity, type=vocab_type, q=f'id:"{term}"'
             ).to_dict()
+            hits = vocabulary_result.get("hits", {}).get("total", 0)
+            current_app.logger.debug(
+                f"[inspire_id={inspire_id}] Vocabulary search returned {hits} results for '{term}' in '{vocab_type}'"
+            )
             return vocabulary_result
         except RequestError as e:
             current_app.logger.warning(
-                f"Error occurred when searching for '{term}' in the vocabulary '{vocab_type}'. INSPIRE record id: {self.inspire_metadata.get('control_number')}. Error: {e}."
+                f"[inspire_id={self.inspire_metadata.get('control_number')}] Error occurred when searching for '{term}' in the vocabulary '{vocab_type}'. INSPIRE record id: {self.inspire_metadata.get('control_number')}. Error: {e}."
             )
         except NoResultFound:
             current_app.logger.warning(
-                f"No result found when searching for '{term}' in the vocabulary '{vocab_type}'. INSPIRE record id: {self.inspire_metadata.get('control_number')}. Error: {e}."
+                f"[inspire_id={self.inspire_metadata.get('control_number')}] No result found when searching for '{term}' in the vocabulary '{vocab_type}'. INSPIRE record id: {self.inspire_metadata.get('control_number')}. Error: {e}."
             )
 
     def _transform_accelerators(self, inspire_accelerators):
         """Map accelerators to CDS-RDM vocabulary."""
         mapped = []
         for accelerator in inspire_accelerators:
+            if accelerator.startswith("CERN-"):
```

Comment:
there is another problem with mapping the accelarators and experiments, I saw in the logs previously the format:
`CERN-<ACCELERATOR>-<EXPERIMENT>`
which complicates things, was this adressed?

Reply:
Hmm, this is a bit more complicated to detect IMHO. Usually you are right, we do have `CERN-<ACCELERATOR>-<EXPERIMENT>` but also there are cases like https://inspirehep.net/api/literature/2892909 where we have:
```
"accelerator_experiments": [
{
"record": {
"$ref": "https://inspirehep.net/api/experiments/1275752"
},
"legacy_name": "CERN-NA-062"
}
],
```

where NA-062 is an experiment, I guess we want to check all the possible combinations in this case.